tests/e2e/framework.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+)
+
+const (
+	epHealthz = "/healthz"
+	epMetrics = "/metrics"
+)
+
+var (
+	framework *Framework
+)
+
+type Framework struct {
+	KsmClient *KSMClient
+}
+
+func NewFramework(ksmurl string) (*Framework, error) {
+	ksmClient, err := NewKSMClient(ksmurl)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Framework{
+		KsmClient: ksmClient,
+	}, nil
+}
+
+type KSMClient struct {
+	endpoint *url.URL
+	client   *http.Client
+}
+
+func NewKSMClient(address string) (*KSMClient, error) {
+	u, err := url.Parse(address)
+	if err != nil {
+		return nil, err
+	}
+	u.Path = strings.TrimRight(u.Path, "/")
+
+	return &KSMClient{
+		endpoint: u,
+		client:   &http.Client{},
+	}, nil
+}
+
+func (k *KSMClient) isHealthz() (bool, error) {
```

Comment:
I would move this to the main_test.go as well. Thanks

Reply:
`KSMClient` is defined in `framework.go` , but its method `isHealthz()` defined in `main_test.go`. This looks a bit strange. I think define all setup related work in `main_test.go` would be more reasonable. WDYT?

Reply:
But the `isHealthz ` is used in  `TestIsHealthz ` only? Anyways its a nit really.

Reply:
We will define other methods of `KSMClient`, if only put `isHealthz` in `framework.go`. It's weird for me 😂 . Maybe it would be better if `xxx_test.go` only contained `TestXXX` functions.

Anyway, we can change the layout of code to make it more appropriate any time :)